### PR TITLE
Sonar codemod for removing unused local variable

### PR DESCRIPTION
**Pull request by carlosu7**
Codemod to remove unused local variables which expression is a variable (`NameExpr`) or just a Literal expression like a single boolean, char, double, integer, long, null, string or a text block string. We are not considering create object expression, method call expressions, condition expressions, etc. because all of them have an expression node and that expression node could result in a method call expression where a process could be performed and deleting it could result on some unexpected behaviors.

Examples of unused `LiteralExpr` : 
* String msg = "my message"
* char myChar = 'c';
* int ten = 10;
* etc

sonar cloud report example: https://sonarcloud.io/project/issues?fileUuids=AYvtrjqJLCzGLicz7Abq&resolved=false&types=CODE_SMELL&id=nahsra_WebGoat_10_23

reference: https://rules.sonarsource.com/java/RSPEC-1481/

### CODE REVIEW

File: core-codemods/src/main/java/io/codemodder/codemods/RemoveUnusedLocalVariableCodemod.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  public boolean onIssueFound(
+      final CodemodInvocationContext context,
+      final CompilationUnit cu,
+      final VariableDeclarator variableDeclarator,
+      final Issue issue) {
+
+    final Optional<Expression> initializer = variableDeclarator.getInitializer();
+
+    if (initializer.isPresent()) {
+      final Expression initializerExpr = initializer.get();
```
Comment (nahsra) [inline core-codemods/src/main/java/io/codemodder/codemods/RemoveUnusedLocalVariableCodemod.java:46]: Shouldn't the `return true` be line 54 or 55? In all other cases, nothing is removed.

File: core-codemods/src/main/resources/io/codemodder/codemods/RemoveUnusedLocalVariableCodemod/description.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+TODO
```
Comment (nahsra) [inline core-codemods/src/main/resources/io/codemodder/codemods/RemoveUnusedLocalVariableCodemod/description.md:1]:
```
This change removes unused variables. Unused variables make the code harder to read, which will lead to confusion and bugs. We only remove variables that have no state-changing effects.
```

File: core-codemods/src/main/java/io/codemodder/codemods/RemoveUnusedLocalVariableCodemod.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+      if (initializerExpr instanceof LiteralExpr || initializerExpr instanceof NameExpr) {
+        final Optional<Node> variableDeclarationExprOptional = variableDeclarator.getParentNode();
+
+        if (variableDeclarationExprOptional.isPresent()) {
```
Comment (nahsra) [inline core-codemods/src/main/java/io/codemodder/codemods/RemoveUnusedLocalVariableCodemod.java:51]: Does this handle the case where:

```
int a =1, b=2;
```

Where only `a` is unused?

File: core-codemods/src/main/java/io/codemodder/codemods/RemoveUnusedLocalVariableCodemod.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+@Codemod(
+    id = "sonar:java/remove-unused-local-variable-s1481",
+    reviewGuidance = ReviewGuidance.MERGE_WITHOUT_REVIEW,
+    executionPriority = CodemodExecutionPriority.HIGH)
+public final class RemoveUnusedLocalVariableCodemod
+    extends SonarPluginJavaParserChanger<VariableDeclarator> {
+
+  @Inject
+  public RemoveUnusedLocalVariableCodemod(
+      @ProvidedSonarScan(ruleId = "java:S1481") final RuleIssues issues) {
```
Comment (nahsra) [inline core-codemods/src/main/java/io/codemodder/codemods/RemoveUnusedLocalVariableCodemod.java:32]: We always seem to use `MATCHES_START`, so I think we should change `SonarPluginJavaParserChanger` to, by default, use `RegionNodeMatcher.MATCHES_START` without the user having to write this boilerplate every time. Then, if an individual codemod wants to change it, we can let them use this signature here to override that.

**Review by nahsra — CHANGES_REQUESTED** [review]

File: core-codemods/src/main/java/io/codemodder/codemods/RemoveUnusedLocalVariableCodemod.java (hunk 3 above)
Comment (carlosu7) [reply core-codemods/src/main/java/io/codemodder/codemods/RemoveUnusedLocalVariableCodemod.java:51]: no, i will add a test case and refactor code considering that scenario

**Review by nahsra — APPROVED** [review]

**Pull request merged**